tests/validation/tests/v3_api/test_cluster_templates.py

```diff
@@ -0,0 +1,448 @@
+
+    # client = get_admin_client()
+    standard_user_client = Client(url=CATTLE_API_URL,
+                                  token=user_token["user_standard"]["token"],
+                                  verify=False)
+    cluster_template = standard_user_client.create_cluster_template(
+            name=random_test_name("template"), description="test-template")
+    clusterTemplateId = cluster_template.id
+
+    cluster_template_revision = \
+        standard_user_client.create_cluster_template_revision(
+            name=random_test_name("revision"),
+            clusterConfig=cluster_config,
+            clusterTemplateId=clusterTemplateId,
+            enabled="true", questions=questions)
+    cluster = create_node_cluster(
+                standard_user_client, name=random_test_name("test-auto"),
+                clusterTemplateRevisionId=cluster_template_revision.id,
+                answers=answers)
+
+    # Verify that the cluster's applied spec has the parameters set as expected
+
+    assert cluster.appliedSpec.rancherKubernetesEngineConfig.\
+        kubernetesVersion == "v1.15.4-rancher1-2"
+    assert cluster.appliedSpec.rancherKubernetesEngineConfig.services.etcd.\
+        backupConfig.s3BackupConfig.bucketName == RANCHER_S3_BUCKETNAME
+    assert cluster.appliedSpec.rancherKubernetesEngineConfig.services.\
+        etcd.backupConfig.s3BackupConfig.endpoint == RANCHER_S3_ENDPOINT
+    assert cluster.appliedSpec.rancherKubernetesEngineConfig.services.etcd.\
+        backupConfig.s3BackupConfig.accessKey == AWS_ACCESS_KEY_ID
+    assert cluster.appliedSpec.rancherKubernetesEngineConfig.services.etcd.\
+        backupConfig.s3BackupConfig.type == "/v3/schemas/s3BackupConfig"
+    assert cluster.appliedSpec.rancherKubernetesEngineConfig.network.plugin ==\
+        "flannel"
+
+    # Verify flannel pod in the kube-system namespace
+    cmd = "get pods -l k8s-app=flannel --namespace kube-system"
+    pod_result = execute_kubectl_cmd(cmd)
+
+    assert (len(["items"])) == 1
+
+    for pod in pod_result["items"]:
+        print(pod["metadata"]["name"])
+        assert "flannel" in (pod["metadata"]["name"])
+
+    # Perform Backup
+    backup = cluster.backupEtcd()
+    backupname = backup['metadata']['name']
+    etcdbackups = cluster.etcdBackups(name=backupname)
+    etcdbackupdata = etcdbackups['data']
+    backupId = etcdbackupdata[0]['id']
+    print("BackupId", backupId)
+    wait_for_backup_to_active(standard_user_client, cluster, backupname)
+
+    cluster_cleanup(standard_user_client, cluster)
+
+
+def test_cluster_template_create_edit_adminuser():
+
+    client = get_admin_client()
+    cluster_template_create_edit(client)
+
+
+def test_cluster_template_create_edit_stduser():
+
+    standard_user_client = Client(
+                         url=CATTLE_API_URL,
+                         token=user_token["user_standard"]["token"],
+                         verify=False)
+    cluster_template_create_edit(standard_user_client)
+
+
```

Comment:
Add test description that outlines the workflow

Reply:
Done